# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/d3d/DisplayD3D.cpp
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/renderer/d3d/DisplayD3D.cpp
--- a/gfx/angle/checkout/src/libANGLE/renderer/d3d/DisplayD3D.cpp	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/d3d/DisplayD3D.cpp	Fri Nov 08 17:06:00 2019 +0000
@@ -215,6 +215,15 @@
     return mRenderer->createStreamProducerD3DTexture(consumerType, attribs);
 }
 
+ExternalImageSiblingImpl *DisplayD3D::createExternalImageSibling(const gl::Context *context,
+                                                                 EGLenum target,
+                                                                 EGLClientBuffer buffer,
+                                                                 const egl::AttributeMap &attribs)
+{
+    ASSERT(mRenderer != nullptr);
+    return mRenderer->createExternalImageSibling(context, target, buffer, attribs);
+}
+
 egl::Error DisplayD3D::makeCurrent(egl::Surface *drawSurface,
                                    egl::Surface *readSurface,
                                    gl::Context *context)
@@ -290,14 +299,33 @@
                                                   attribs);
 
         case EGL_D3D_TEXTURE_ANGLE:
-            return mRenderer->getD3DTextureInfo(
-                configuration, static_cast<IUnknown *>(clientBuffer), nullptr, nullptr, nullptr);
+            return mRenderer->getD3DTextureInfo(configuration,
+                                                static_cast<IUnknown *>(clientBuffer), nullptr,
+                                                nullptr, nullptr, nullptr);
 
         default:
             return DisplayImpl::validateClientBuffer(configuration, buftype, clientBuffer, attribs);
     }
 }
 
+egl::Error DisplayD3D::validateImageClientBuffer(const gl::Context *context,
+                                                 EGLenum target,
+                                                 EGLClientBuffer clientBuffer,
+                                                 const egl::AttributeMap &attribs) const
+{
+    switch (target)
+    {
+        case EGL_D3D11_TEXTURE_ANGLE:
+        {
+            return mRenderer->getD3DTextureInfo(nullptr, static_cast<IUnknown *>(clientBuffer),
+                                                nullptr, nullptr, nullptr, nullptr);
+        }
+
+        default:
+            return DisplayImpl::validateImageClientBuffer(context, target, clientBuffer, attribs);
+    }
+}
+
 void DisplayD3D::generateExtensions(egl::DisplayExtensions *outExtensions) const
 {
     mRenderer->generateDisplayExtensions(outExtensions);
@@ -358,6 +386,11 @@
     return mRenderer->getMaxSupportedESVersion();
 }
 
+gl::Version DisplayD3D::getMaxConformantESVersion() const
+{
+    return mRenderer->getMaxConformantESVersion();
+}
+
 void DisplayD3D::handleResult(HRESULT hr,
                               const char *message,
                               const char *file,
@@ -372,4 +405,10 @@
 
     mStoredErrorString = errorStream.str();
 }
+
+void DisplayD3D::populateFeatureList(angle::FeatureList *features)
+{
+    mRenderer->getFeatures().populateFeatureList(features);
+}
+
 }  // namespace rx